Cabal/tests/PackageTests/Tests.hs

```diff
@@ -0,0 +1,157 @@
+module PackageTests.Tests(tests) where
+
+import PackageTests.PackageTester
+
+import qualified PackageTests.BenchmarkStanza.Check
+import qualified PackageTests.TestStanza.Check
+import qualified PackageTests.DeterministicAr.Check
+import qualified PackageTests.TestSuiteTests.ExeV10.Check
+
+import Distribution.Simple.Utils (withFileContents)
+import Control.Monad
+
+import Data.Version
+import Test.Tasty (TestTree, testGroup, mkTimeout, localOption)
+import Test.Tasty.HUnit (testCase)
+
+-- TODO: turn this into a "test-defining writer monad".
+-- This will let us handle scoping gracefully.
+tests :: SuiteConfig -> [TestTree]
+tests config =
+  tail [ undefined
+
+  -- External tests
+  , tc "BenchmarkStanza"  PackageTests.BenchmarkStanza.Check.suite
+  , tc "TestStanza"       PackageTests.TestStanza.Check.suite
+  , tc "DeterministicAr"  PackageTests.DeterministicAr.Check.suite
+
+  -- Test suite tests
+  , testGroup "TestSuiteTests"
+    [ testGroup "ExeV10"
+      (PackageTests.TestSuiteTests.ExeV10.Check.tests config)
+    , testGroup "LibV09" $
+      let
+        tcs :: FilePath -> TestM a -> TestTree
+        tcs name m
+            = testCase name (runTestM config ("TestSuiteTests/LibV09")
+                                             (Just name) m)
+      in [ tcs "Build" $ cabal_build ["--enable-tests"]
+         , localOption (mkTimeout $ 10 ^ (8 :: Int))
+         . tcs "Deadlock" $ do
+            cabal_build ["--enable-tests"]
+            shouldFail $ cabal "test" []
+         ]
+    ]
+
+  -- Inline tests
+  , tc "BenchmarkExeV10" $ cabal_build ["--enable-benchmarks"]
+
+  , tc "BenchmarkOptions" $ do
```

Comment:
I think we should leave this for another PR; when I originally decided to port these tests, my plan was to shorten the tests but otherwise not change them at all. Most of the tests that call cabal_build and then some commands originally (and still) were *solely* testing if ./Setup ran without exiting non-zero.

Actually, originally, the test suite was a bit more like you suggested: you had to explicitly assert that a given command had succeeded or failed. I hated this, because there were lots of tests which outright FORGOT to test the exit code, and it was a pain for "setup" operations, where you implicitly did not want it to fail.

I'd like to avoid boilerplate. If by and large most tests are `successfulExitOnly` and `noSetup`, can we just keep that implicit?